Description: Get src/db.cpp to build with the version of libdb in Squeeze
 The DB_LOG_AUTO_REMOVE and DB_LOG_IN_MEMORY DB_LOG_IN_MEMORY defines do
 not exist.  Use DB_LOG_AUTOREMOVE and DB_LOG_INMEMORY instead.
Author: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2012-12-21

--- bitcoin-0.7.2.orig/src/db.cpp
+++ bitcoin-0.7.2/src/db.cpp
@@ -87,7 +87,11 @@ bool CDBEnv::Open(boost::filesystem::pat
     dbenv.set_errfile(fopen(pathErrorFile.string().c_str(), "a")); /// debug
     dbenv.set_flags(DB_AUTO_COMMIT, 1);
     dbenv.set_flags(DB_TXN_WRITE_NOSYNC, 1);
+#ifdef DB_LOG_AUTOREMOVE
+    dbenv.set_flags(DB_LOG_AUTOREMOVE, 1);
+#else
     dbenv.log_set_config(DB_LOG_AUTO_REMOVE, 1);
+#endif
     int ret = dbenv.open(strPath.c_str(),
                      DB_CREATE     |
                      DB_INIT_LOCK  |
@@ -122,7 +126,11 @@ void CDBEnv::MakeMock()
     dbenv.set_lk_max_locks(10000);
     dbenv.set_lk_max_objects(10000);
     dbenv.set_flags(DB_AUTO_COMMIT, 1);
+#ifdef DB_LOG_INMEMORY
+    dbenv.set_flags(DB_LOG_INMEMORY, 1);
+#else
     dbenv.log_set_config(DB_LOG_IN_MEMORY, 1);
+#endif
     int ret = dbenv.open(NULL,
                      DB_CREATE     |
                      DB_INIT_LOCK  |
